### PR TITLE
[front] - refacto(DocumentOrTableUploadOrEditModal): enhancements & fixes

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the `DocumentOrTableUploadOrEditModal` component, splitting it into separate `DocumentUploadOrEditModal` and `TableUploadOrEditModal` components. It enhances the user experience by adding input validation, improving error handling, and optimizing the loading states for both document and table uploads.

**References:**
 - https://github.com/dust-tt/dust/issues/7632

## Risk

Mainly moving stuff around but should be tested on front-edge first

## Deploy Plan

- Deploy `front-edge`
- Deploy `front`

### CODE REVIEW

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx [hunk 1]
```diff
@@ -48,123 +58,448 @@ interface DocumentOrTableUploadOrEditModalProps {
...
+    disabled: !isTable,
+  });
+
+  const { document, isDocumentError, isDocumentLoading } =
+    useDataSourceViewDocument({
+      owner: props.owner,
+      dataSourceView: props.dataSourceView,
+      documentId: !isTable ? initialId ?? null : null,
+      disabled: isTable,
+    });
```
Comment (Fraggle) [inline front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:80]: Why aren't theses hooks not directly in the respective components ?

Comment (Fraggle) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:80]: It looks like you just need to pass dataSourceView, initialId and owner to the 2 components.

File: front/lib/swr/data_source_views.ts [hunk 2]
```diff
@@ -174,25 +174,27 @@ export function useDataSourceViewDocument({
...
-  const disabled = !dataSourceView || !documentId;
+  const isDisabled = !dataSourceView || !documentId || disabled;
 
   const { data, error, mutate } = useSWRWithDefaults(
-    disabled
+    isDisabled
       ? null
       : `/api/w/${owner.sId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/documents/${encodeURIComponent(documentId)}`,
     dataSourceViewDocumentFetcher
   );
```
Comment (Fraggle) [inline front/lib/swr/data_source_views.ts:193]: Now that you have a real disabled, I think it's more solid to distinguish between being able to compute an url and disabling actively like this:

```ts
const url = dataSourceView && documentId ? `/api/w/${owner.sId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/documents/${encodeURIComponent(documentId)}` : null;

const { data, error, mutate } = useSWRWithDefaults(
 url,
 dataSourceViewDocumentFetcher,
 { disabled }
);
```

**Review by Fraggle — APPROVED** [review]
LGTM :)

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx]: Very fair 👍🏼

**Pull request merged**